orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/AbstractDeployStrategyStage.groovy

```diff
@@ -65,13 +65,13 @@ abstract class AbstractDeployStrategyStage extends AbstractCloudProviderAwareSta
       .withTask("determineSourceServerGroup", DetermineSourceServerGroupTask)
       .withTask("determineHealthProviders", DetermineHealthProvidersTask)
 
+    correctContext(stage)
```

Comment:
this needs to happen before the deployStagePreProcessors because they (the only impl being the AWS one) expects the corrected context as it does a `stage.mapTo(StageData)` to check the `supports` call